### PR TITLE
Replace microns/micrometers (µm) w/ 0.1mm

### PR DESCRIPTION
**Pull request by classicrocker883**
### Description
A micron/micometer is 1/1000 of a millimeter, not 1/10.

Correct me if I'm wrong but aren't these values measured in 1/10th of a millimeter, not Microns?

Unless 
The actual unit denotation should be (*dmm*) - decimillimeter, but `0.1mm` or perhaps `1/10mm` makes more sense.

### Related Issues
in **Marlin\src\feature\probe_temp_comp.cpp**
I am a bit confused by this, something doesn't add up:
```y
  // offset in µm   -   changed to 0.1mm
  float offset = 0.0f;
...
  // convert offset to mm and apply it
  meas_z -= offset / 1000.0f;
```
if `offset` is 1/10mm, to convert to mm then that would be " * 10 ".
so if this were actually in microns, then dividing by 1000 you get nanometers.

### CODE REVIEW

**Comment by gudvinr** [comment]
> The actual unit denotation should be (dmm) - decimillimeter

I am pretty sure no such thing exists. You can't use multiple SI prefixes for single units.

**Comment by thisiskeithb** [comment]
This is related to work done in https://github.com/MarlinFirmware/Marlin/pull/23033 by @tombrazier.

**Comment by classicrocker883** [comment]
> > The actual unit denotation should be (dmm) - decimillimeter
> 
> I am pretty sure no such thing exists. You can't use multiple SI prefixes for single units.

https://www.wikidata.org/wiki/Q20706221

from https://www.ibiblio.org/units/dictD.html
>decimillimeter (dmm)
a metric unit of distance equal to 0.1 millimeter (10-4 meter) or about 3.937 [mils](https://www.ibiblio.org/units/dictM.html#mil). The unit is used in civil engineering for stating the results of penetration tests of asphalt concrete, in which a needle is pushed into the concrete under specified conditions. Although this unit is allowed by some standards agencies, the use of compound prefixes such as decimilli- is not permitted in the [SI](https://www.ibiblio.org/units/sipm.html).

**Comment by tombrazier** [comment]
This code predates me but looking at it I cannot see any suggestion that the values are tenths of a mm. It is treated as µm at the end of the function where conversion to mm involves a division by 1000. And the values are set by M871 for which the documentation states µm.

**Comment by classicrocker883** [comment]
@thisiskeithb why was this closed? 

clearly µm are incorrect. even so in the documentation

**Comment by classicrocker883** [comment]
> This code predates me but looking at it I cannot see any suggestion that the values are tenths of a mm. It is treated as µm at the end of the function where conversion to mm involves a division by 1000. And the values are set by M871 for which the documentation states µm.

converting to µm involves division by 1000, not the other way around. and just because M871 documentation states µm doesn't mean that it is so. i've made a pull request to make those changes on the docs. because i thought using `M871 V` the values used were small, in 1/10 of a mm, not 100 times that.

@tombrazier so do you know if these values are actually in microns or something else?
- `PTC_PROBE_ZOFFS` / `z_offsets_probe`
- `PTC_BED_ZOFFS` / `z_offsets_bed`
- `PTC_HOTEND_ZOFFS` / `z_offsets_hotend`

basically asking, are these numbers in tenths, hundredths, thousandths?

**Comment by tombrazier** [comment]
1mm is 1000µm. I.e. divide by 1000 to get mm.

What is it that makes you think the value is in tenths of a mm?

**Comment by tombrazier** [comment]
The values for PTC_PROBE_ZOFFS, etc. are in µm according to
Configuration_adv.h.
